lesson02/tests/test_job1_main.py

```diff
@@ -0,0 +1,103 @@
+"""
+Tests for job1 main.py
+"""
+import sys
+from unittest import TestCase, mock
+sys.path.append("/Users/olegr/work/_mine/DataEngineering/py/DE2022_my/lesson02/job1")
+from job1 import main
+
+
+class Job1MainFunctionTestCase(TestCase):
+    """
+    Main tests for job1
+    """
+    print("******* job1 test main:")
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        main.app.testing = True
+        cls.client = main.app.test_client()
+
+    def test_return_code(self):
+        """
+        Raise 400 HTTP code when no 'date' and/or 'path' param
+        """
+        resp = self.client.post(
+            '/',
+            json={
+                'raw_dir': '/foo/bar/',
+                # no 'date' set
+            },
+        )
+        resp2 = self.client.post(
+            '/',
+            json={
+                'date': '2000-01-01',
+                # no 'path' set
+            },
+        )
+        resp3 = self.client.post(
+            '/',
+            json={
+                'date': '3000-01-01',
+                'raw_dir': '/foo/bar/'
+            },
+        )
+        resp4 = self.client.post(
+            '/',
+            json={
+                'date': '202a-01-01',
+                'raw_dir': '/foo/bar/'
+            },
+        )
+        self.assertEqual(400, resp.status_code)
+        self.assertEqual(400, resp2.status_code)
+        self.assertEqual(404, resp3.status_code)
+        self.assertEqual(501, resp4.status_code)
```

Comment:
The disadvantage of this approach is when you've got an exception in resp1, resp2 etc won't work. So it would be difficult to find out what is going wrong in the code. 

The rule of thumb should be "one assert in one test method".
Exception - using of sub-testing: https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests